### PR TITLE
[esl-carousel] inner API update + fixes

### PR DESCRIPTION
**Pull request by ala-n**

### CODE REVIEW

File: src/modules/esl-carousel/core/esl-carousel.utils.ts [hunk 1]
```diff
@@ -111,12 +100,12 @@ function resolveGroupIndex(indexStr: string | ESLCarouselNavIndex, cfg: ESLCarou
   }
   // TODO: extend navigation boundaries
   if (value === -1 && cfg.activeIndex < cfg.count && cfg.activeIndex > 0) {
-    return {index: 0, direction: direction || 'prev'};
+    return {index: 0, direction: direction || ESLCarouselDirection.PREV};
   }
   if (value === 1 && normalize(cfg.activeIndex + cfg.count, cfg.size) > cfg.size - cfg.count) {
```
Comment (NastaLeo) [inline src/modules/esl-carousel/core/esl-carousel.utils.ts:105]: should we here replace normalize by normalizeIndex?

**Review by NastaLeo — APPROVED** [review]

File: src/modules/esl-carousel/core/esl-carousel.utils.ts (hunk 1 above)
Comment (ala-n) [reply src/modules/esl-carousel/core/esl-carousel.utils.ts:105]: Probably in that check we shouldn't.
That's boundary check for groups so it is respect non-loop carousel boundaries by design. Not sure we should do something with conditions for that.

File: src/modules/esl-carousel/renderers/esl-carousel.default.renderer.ts [hunk 2]
```diff
@@ -77,53 +77,57 @@ export class ESLDefaultCarouselRenderer extends ESLCarouselRenderer {
     this.$area.style.transform = `translate3d(${this.vertical ? `0px, ${offset}px` : `${offset}px, 0px`}, 0px)`;
   }
 
-  /** Animate scene offset */
-  protected async animateTransformOffset(offset: number = -this.getOffset(this.currentIndex)): Promise<void> {
+  /** Animate scene offset to index */
```
Comment (dshovchko) [inline src/modules/esl-carousel/renderers/esl-carousel.default.renderer.ts:80]:
```suggestion
  /** Animates scene offset to index */
```

File: src/modules/esl-carousel/core/esl-carousel.types.ts [hunk 3]
```diff
@@ -47,5 +52,15 @@ export type ESLCarouselNavInfo = {
   /** Target index */
   index: number;
   /** Direction to reach the index */
-  direction: ESLCarouselDirection | null;
+  direction?: ESLCarouselDirection;
 };
+
+/** {@link ESLCarousel} action params interface */
+export interface ESLCarouselActionParams {
+  /** Element requester of the change */
```
Comment (dshovchko) [inline src/modules/esl-carousel/core/esl-carousel.types.ts:60]:
```suggestion
  /** Element that requests changes */
```

File: src/modules/esl-carousel/core/esl-carousel.types.ts [hunk 4]
```diff
@@ -47,5 +52,15 @@ export type ESLCarouselNavInfo = {
...
+  activator?: any;
+  /** Direction to move to. */
```
Comment (dshovchko) [inline src/modules/esl-carousel/core/esl-carousel.types.ts:62]:
```suggestion
  /** Direction to move to */
```

File: src/modules/esl-carousel/core/esl-carousel.ts [hunk 5]
```diff
@@ -298,12 +294,28 @@ export class ESLCarousel extends ESLBaseElement {
...
+    this.renderer.move(offset, from, this.mergeParams(params));
+  }
+
+  /** Commit slides to the nearest stable position */
+  public commit(offset: number, from: number = this.activeIndex, params: Partial<ESLCarouselActionParams> = {}): Promise<void> {
+    if (!this.renderer) return Promise.reject();
+    return this.renderer.commit(offset, from, this.mergeParams(params));
+  }
+
+  /** Merge request params with default params */
```
Comment (dshovchko) [inline src/modules/esl-carousel/core/esl-carousel.ts:315]:
```suggestion
  /** Merges request params with default params */
```

File: src/modules/esl-carousel/core/esl-carousel.ts [hunk 6]
```diff
@@ -298,12 +294,28 @@ export class ESLCarousel extends ESLBaseElement {
...
+  public goTo(target: HTMLElement | ESLCarouselSlideTarget, params: Partial<ESLCarouselActionParams> = {}): Promise<void> {
     if (target instanceof HTMLElement) return this.goTo(this.indexOf(target), params);
     if (!this.renderer) return Promise.reject();
-    const nav = toIndex(target, this.state);
-    const direction = (this.loop ? params.direction : null) || nav.direction || 'next';
-    return this.renderer.navigate(nav.index, direction, params);
+    return this.renderer.navigate(toIndex(target, this.state), this.mergeParams(params));
+  }
+
+  /** Move slides by the passed offset */
```
Comment (dshovchko) [inline src/modules/esl-carousel/core/esl-carousel.ts:303]:
```suggestion
  /** Moves slides by the passed offset */
```

File: src/modules/esl-carousel/core/esl-carousel.ts [hunk 7]
```diff
@@ -298,12 +294,28 @@ export class ESLCarousel extends ESLBaseElement {
...
+  public move(offset: number, from: number = this.activeIndex, params: Partial<ESLCarouselActionParams> = {}): void {
+    if (!this.renderer) return;
+    this.renderer.move(offset, from, this.mergeParams(params));
+  }
+
+  /** Commit slides to the nearest stable position */
```
Comment (dshovchko) [inline src/modules/esl-carousel/core/esl-carousel.ts:309]:
```suggestion
  /** Commits slides to the nearest stable position */
```

File: src/modules/esl-carousel/core/esl-carousel.slide.ts [hunk 8]
```diff
@@ -59,7 +60,9 @@ export class ESLCarouselSlide extends ESLMixinElement {
   }
 
   protected override disconnectedCallback(): void {
-    this.$carousel?.removeSlide && this.$carousel.removeSlide(this.$host);
+    // A disconnected callback is not directly related to slide removal from the carousel
+    // e.g. carousel itself can be removed from the DOM so child slides behaves accordingly
```
Comment (dshovchko) [inline src/modules/esl-carousel/core/esl-carousel.slide.ts:64]:
```suggestion
    // e.g. carousel itself can be removed from the DOM so child slides behave accordingly
```

**Review by dshovchko — APPROVED** [review]

**Review by fshovchko — APPROVED** [review]

**Pull request merged**